release notes/v0.42.0.md

```diff
@@ -0,0 +1,165 @@
+
+```javascript
+import { WebSocket } from "k6/experimental/websockets";
+import {
+  setTimeout,
+  clearTimeout,
+  setInterval,
+  clearInterval
+} from "k6/experimental/timers";
+
+const CLOSED_STATE = 3
+
+export default function () {
+  const params = {
+    "tags": {
+      "my_tag": "hello"
+    }
+  };
+
+  const ws = new WebSocket('ws://localhost:10000', null, params);
+
+  ws.onopen = () => {
+    console.log('connected');
+    ws.send(Date.now().toString());
+  };
+
+  let intervalId = setInterval(() => {
+    ws.ping();
+    console.log("Pinging every 1 sec (setInterval test)");
+  }, 1000);
+
+  let timeout1id = setTimeout(function () {
+    console.log('2 seconds passed, closing the socket');
+    clearInterval(intervalId);
+    ws.close();
+  }, 2000);
+
+  ws.onclose = () => {
+    clearTimeout(timeout1id);
+    console.log('disconnected');
+  };
+
+  ws.onping = () => {
+    console.log("PING!");
+  };
+
+  ws.onpong = () => {
+    console.log("PONG!");
+  };
+
+  // Multiple event handlers on the same event
+  ws.addEventListener("pong", () => {
+    console.log("OTHER PONG!");
+  });
+
+  ws.onmessage = (m) => {
+    let parsed = parseInt(m.data, 10)
+    if (Number.isNaN(parsed)) {
+      console.log('Not a number received: ', m.data);
+      return
+    }
+
+    console.log(`Roundtrip time: ${Date.now() - parsed} ms`);
+
+    let timeoutId = setTimeout(function () {
+      if (ws.readyState == CLOSED_STATE) {
+        console.log("Socket closed, not sending anything");
+
+        clearTimeout(timeoutId);
+        return;
+      }
+
+      ws.send(Date.now().toString());
+    }, 500);
+  };
+
+  ws.onerror = (e) => {
+    if (e.error != "websocket: close sent") {
+      console.log('An unexpected error occurred: ', e.error);
+    }
+  };
+};
+```
+</details>
+
+The [module docs](https://k6.io/docs/javascript-api/k6-experimental/websockets) has a complete reference, and some examples.
+
+### Wildcard support for `hosts` option [#2747](https://github.com/grafana/k6/pull/2747)
+
+Thanks to the great effort from @eugercek, the [hosts](https://k6.io/docs/using-k6/k6-options/reference/#hosts) option now accepts domains that contain a wildcard at the beginning.
+It can be helpful for setting multiple subdomains of the same domain, so instead of setting `subdomain1.k6.io': '1.2.3.4', 'subdomain2.k6.io': '1.2.3.4'` it is possible to use the wildcard for setting directly `*.k6.io: '1.2.3.4'`.
+
+```js
+export const options = {
+  hosts: {
+    '*.k6.io': '1.2.3.4',
+  },
+}
+```
+
+### Enhancements, and UX improvements
+
+- [#2660](https://github.com/grafana/k6/pull/2660) Pre-loads the operating system TLS certificates. Thanks, @tbourrely.
+- [#2773](https://github.com/grafana/k6/pull/2773) Ensures that JavaScript runtime makes only one copy of the exports for each module including built-in ones.
```

Comment:
Shouldn't this be in the bugfixes section?

Reply:
It seems it was an undefined behaviour before so I interpret it as an improvement. I defer to @mstoykov.

Reply:
I am pretty sure it was a minor bug. Not so much "undefined behavior", rather it was "unintended" :smile: